packages/legend-query/src/components/QueryBuilderProjectionPanel.tsx

```diff
@@ -494,31 +517,64 @@ export const QueryBuilderProjectionPanel = observer(
       (columnState) => columnState.isBeingDragged,
     );
     const handleDrop = useCallback(
-      (item: QueryBuilderExplorerTreeDragSource): void =>
-        projectionState.addColumn(
-          new QueryBuilderSimpleProjectionColumnState(
-            projectionState,
-            buildPropertyExpressionFromExplorerTreeNodeData(
-              queryBuilderState.explorerState.nonNullableTreeData,
-              item.node,
-              projectionState.queryBuilderState.graphManagerState.graph,
+      (
+        item:
+          | QueryBuilderExplorerTreeDragSource
+          | QueryBuilderFunctionsExplorerDragSource,
+        type: string,
+      ): void => {
+        if (type === QUERY_BUILDER_FUNCTIONS_EXPLORER_TREE_DND_TYPE.FUNCTION) {
+          const derivationProjectionColumn =
+            new QueryBuilderDerivationProjectionColumnState(
+              projectionState,
+              new RawLambda(
+                [{ _type: 'var', name: 'x' }],
+                [
+                  {
+                    _type: 'string',
+                    multiplicity: { lowerBound: 1, upperBound: 1 },
+                    values: [''],
+                  },
+                ],
+              ),
+            );
+          derivationProjectionColumn.derivationLambdaEditorState.setLambdaString(
+            `x|${generateFunctionLambdaString(
+              (item as QueryBuilderFunctionsExplorerDragSource).node
+                .packageableElement as ConcreteFunctionDefinition,
+            )}`,
+          );
+          projectionState.addColumn(derivationProjectionColumn);
+        } else {
```

Comment:
--> so assuming in the `else` block like this is not wrong but could be dangerous in the future when we add more handling, perhaps it's better to rewrite this as a `switch-case` here